datafusion/physical-plan/src/aggregates/group_values/row.rs

```diff
@@ -170,15 +213,40 @@ impl GroupValues for GroupValuesRows {
             .take()
             .expect("Can not emit from empty rows");
 
+        let map_contents = self.var_map.take().into_state();
+        let map_contents = map_contents.as_string::<i32>();
+
         let mut output = match emit_to {
             EmitTo::All => {
-                let output = self.row_converter.convert_rows(&group_values)?;
+                let mut output = self.row_converter.convert_rows(&group_values)?;
+
+
+                // Index Array: [0, 1, 1, 0]
+                // Data Array: ['a', 'c']
```

Comment:
This code copies all the strings again, which likely takes significant time

One way to make this faster is likey to  special case the "has no nulls" path (so we can avoid `if let Some(.)` check each row

The only other ways I can think to make this faster is to avoid copying the strings all together. The only way to do so I can figure are:
1. Return a DictionaryArray as output 
4. Return a `StringViewArray` (similar to DictionayArray) https://github.com/apache/datafusion/issues/10918

🤔 

Maybe we can try to hack in the ability to have the HashAggregateExec return `Dictionary(Int32, String)` for these multi-column groups and see if we can show significant performance improvements

If so then we can figure out how to thread that ability through the engine.